### PR TITLE
upgraded python version used in tests

### PR DESCRIPTION
**Pull request by sredxny**
## Description

- Upgraded version of python to 3.9 and handled flaky tests (skip some of them) as we do in master.
- Added github flow to lint the new code and removed the go vet from ci tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4532/merge
Commit: 1018dabfa26a13ff2c16c2315db684cc522318a8 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3689449351)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4532/merge
Commit: 2ff8c9a535531dd8c1700f2c1727bf220752d0a9 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3689794149)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4532/merge
Commit: dda0fccc0ff8041c905359dcad419d88d0eed9db 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3690634590)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4532/merge
Commit: 40fd93c8299d88d1cce2ab9f38df364620a1162d 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3690760531)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4532/merge
Commit: 51af84f61c81871cccb403a382200be43057c936 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3690878744)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4532/merge
Commit: e18cf8d1126083c6c4346d9275a5267a65533b13 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3690848377)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4532/merge
Commit: cc2f75c0e189c3ffa150c26373322994a65f8911 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3690950585)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/4532/merge
Commit: 6862ffa4adbd8e1621bf6e5fea5cc55739a0f748 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3690961308)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4532/merge
Commit: 16d2361c63b1a549064df1d9739ede3213e7c107 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3690996389)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4532/merge
Commit: b4bd8deb032263e3b9b7300196bd7aa1dd5e5850 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3694647412)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4532/merge
Commit: aa9e4d5955427ddcb7d9f593f30966637f1a2d35 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3695476939)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4532/merge
Commit: c4a41ba77af62d47fdde2d0bc05d52a5e8b4f822 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3695492773)

File: storage/redis_controller_test.go [hunk 1]
```diff
@@ -24,7 +27,7 @@ func TestRecoverLoop(t *testing.T) {
 
 	conf := config.Default
 
-	rc := NewRedisController(ctx)
+	rc := NewRedisController()
```
Comment (buger) [inline storage/redis_controller_test.go:30]: So ctx argument is optional here?

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - release-4-lts
+  pull_request:
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
```
Comment (mitjaziv) [inline .github/workflows/golangci-lint.yml:17]: There is already v3 available.

File: .github/workflows/golangci-lint.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
```
Comment (mitjaziv) [inline .github/workflows/golangci-lint.yml:20]: There is already v3 available.

File: storage/redis_controller_test.go (hunk 1 above)
Comment (sredxny) [reply storage/redis_controller_test.go:30]: Not really, if doesn't event receives that value in this branch. In other hand, in master branch the signature for this method is:

`func NewRedisController(ctx context.Context) *RedisController`

So, it wasn't even running 😄 . I didn't change the function `NewRedisController` in order to not make big changes

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4532/merge
Commit: e99a6e1ace2e82fcf0a5572e2a329a84017a3d89 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3697064489)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4532/merge
Commit: 3f2c8c72fc41f98c1caa847d074304d226aeca91 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3697563384)

File: .github/workflows/golangci-lint.yml (hunk 2 above)
Comment (sredxny) [reply .github/workflows/golangci-lint.yml]: done!

**Pull request merged**